### PR TITLE
examples/foc: replace the requested velocity input with a more general setpoint input

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

-  examples/foc: replace the requested velocity input with a more general setpoint input.
This is the inital step to support torque/velocity/position control
- examples/foc: temporarily disable error checking for ADC configuration to pass CI
NOTE: this should be reverted after updating nuttx/boards

## Impact
All example/foc configurations must be updated to use new configuration options

### CODE REVIEW

File: examples/foc/foc_adc.h [hunk 1]
```diff
@@ -66,7 +66,7 @@
 #ifdef CONFIG_EXAMPLES_FOC_SETPOINT_CONST
 #  define SETPOINT_ADC_SCALE   (1)
 #  if CONFIG_EXAMPLES_FOC_SETPOINT_CONST_VALUE == 0
-#    error
+/* #    error */
```
Comment (xiaoxiang781216) [inline examples/foc/foc_adc.h:69]: can we merge the patch in  nuttx first?

Comment (raiden00pl) [reply examples/foc/foc_adc.h:69]: I think that if we merge first the patch in nuttx then CI will be broken until this PR merge.

related PR https://github.com/apache/incubator-nuttx/pull/4744

Comment (xiaoxiang781216) [reply examples/foc/foc_adc.h:69]: Both patch can't pass the CI, should we merge them together?

Comment (raiden00pl) [reply examples/foc/foc_adc.h]: It looks like it's the only possibility.
In that case I remove the last commit, due the fact that CI fails for another reason and please merge this PR together with https://github.com/apache/incubator-nuttx/pull/4744

Comment (xiaoxiang781216) [reply examples/foc/foc_adc.h]: Ok, let's merge both together.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**